apiserver/controllers/controllers.go

```diff
@@ -29,9 +32,15 @@ import (
 
 	"github.com/gorilla/websocket"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 func NewAPIController(r *runner.Runner, auth *auth.Authenticator, hub *wsWriter.Hub) (*APIController, error) {
+	id, err := r.GetControllerID()
```

Comment:
The controller ID could be passed as an argument to ```NewAPIController()``` from ```main()```. It spares us from calling this function, as we already have that info in ```main()```.

Reply:
If I see it correctly we don't have the `id` directly in main, what am I missing ?

Reply:
We have this bit:

https://github.com/cloudbase/garm/blob/main/cmd/garm/main.go#L104-L106

which is defined here: https://github.com/cloudbase/garm/blob/main/cmd/garm/main.go#L51-L61

There is no reason why it could not return the controller info as well as the error. You could then pass that info along to whichever component needs it.